xtra-libp2p/src/endpoint.rs

```diff
@@ -354,7 +354,10 @@ impl Endpoint {
             },
         );
 
-        self.controls.insert(peer_id, (control, tasks));
+        if self.controls.insert(peer_id, (control, tasks)).is_some() {
+            tracing::debug!(%peer_id, "Missed drop event, replacing old connection")
```

Comment:
Should this be a warning given this generally should not happen, but could if the connection is replaced very fast?